guest/postfilter/postfilter.go

```diff
@@ -0,0 +1,136 @@
+/*
+   Copyright 2023 The Kubernetes Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+// Package postfilter exports an api.PostFilterPlugin to the host.
+package postfilter
+
+import (
+	"runtime"
+
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/api"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/cyclestate"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/imports"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/mem"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/guest/internal/plugin"
+	"sigs.k8s.io/kube-scheduler-wasm-extension/kubernetes/proto/nodetostatus"
+)
+
+// postfilter is the current plugin assigned with SetPlugin.
+var postfilter api.PostFilterPlugin
+
+// SetPlugin should be called in `main` to assign an api.PostFilterPlugin
+// instance.
+//
+// For example:
+//
+//	func main() {
+//		plugin := filterPlugin{}
+//		postfilter.SetPlugin(plugin)
+//		filter.SetPlugin(plugin)
+//	}
+//
+//	type filterPlugin struct{}
+//
+//	func (filterPlugin) PostFilter(state api.CycleState, pod proto.Pod, filteredNodeStatusMap internalpostfilter.NodeToStatusMap) (int32, status *api.Status) {
+//		// Write state you need on Filter
+//	}
+//
+//	func (filterPlugin) Filter(state api.CycleState, pod api.Pod, nodeInfo api.NodeInfo) (status *api.Status) {
+//		var Filter int32
+//		// Derive Filter for the node name using state set on PreFilter!
+//		return Filter, nil
+//	}
+func SetPlugin(postfilterPlugin api.PostFilterPlugin) {
+	if postfilterPlugin == nil {
+		panic("nil postfilterPlugin")
+	}
+	postfilter = postfilterPlugin
+	plugin.MustSet(postfilterPlugin)
+}
+
+// prevent unused lint errors (lint is run with normal go).
+var _ func() uint64 = _postfilter
+
+// _postfilter is only exported to the host.
+//
+//export postfilter
+func _postfilter() uint64 { //nolint
```

Comment:
uint32 is enough? See prefilter for example which also returns status.

Reply:
`(uint64(nominatingMode) << uint64(32)) | uint64(imports.StatusToCode(status))`

Since two values `nominatingMode` and `status` are returned here, I set uint64 as value